test/test_domains.py

```diff
@@ -47,9 +47,9 @@ def test_subclass():
     assert issubclass(Reals[2], Reals)
     assert issubclass(Reals[2], Reals[2])
     assert not issubclass(Reals, Real)
-    assert not issubclass(Reals, Real[2])
+    assert not issubclass(Reals, Reals[2])
```

Comment:
@eb8680 I don't know how tests passed in #352 before this change, but this change was needed to make tests pass in this PR 🤷